c2corg_ui/static/js/utils.js

```diff
@@ -0,0 +1,58 @@
+goog.provide('app.utils');
+
+goog.require('goog.asserts');
+
+
+/**
+ * @param {string} module The document type.
+ * @param {string} documentId The document id.
+ * @param {string} culture Culture.
+ * @return {string} Url.
+ */
+app.utils.buildDocumentUrl = function(module, documentId, culture) {
+  return '/{module}/{id}/{culture}'
+    .replace('{module}', module)
+    .replace('{id}', documentId)
+    .replace('{culture}', culture);
+};
+
+
+/**
+ * Cultures ordered by their relevance.
+ * @const
+ * @type {Array.<string>}
+ */
+app.utils.LANG_PRIORITY = ['fr', 'en', 'it', 'de', 'es', 'ca', 'eu'];
+
+
+/**
+ * Get the "best" locale for a document. If available this is the
+ * `preferedCulture` (usually the interface language), if not the locale in
+ * the "most relevant" culture.
+ *
+ * @param {Object} document The document.
+ * @param {string} preferedCulture The prefered culture
+ * @param {Array.<string>} cultures All cultures.
+ * @return {Object} The "best" culture.
+ */
+app.utils.getBestLocale = function(document, preferedCulture, cultures) {
+  var locales = document['locales'];
+  var availableLocales = {};
+  $.each(locales, function(i, locale) {
+    availableLocales[locale['culture']] = locale;
+  });
+
+  if (availableLocales[preferedCulture] !== undefined) {
+    return availableLocales[preferedCulture];
+  } else {
+    var localeToUse = null;
+    $.each(app.utils.LANG_PRIORITY, function(i, culture) {
```

Comment:
Not seen the return. You can use standard `some()` in this case, and return true.